### PR TITLE
[FLINK-18867][hive] Generic table stored in Hive catalog is incompati…

### PR DESCRIPTION
**Pull request by lirui-apache**
…ble between 1.10 and 1.11

## What is the purpose of the change

Fix the issue that tables created in 1.10 cannot be accessed in 1.11.

## Brief change log

  - Try both new and old key when retrieving table schema from the properties.

## Verifying this change

Manually verified.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
@JingsongLi Please take a look.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogGenericMetadataTest.java [hunk 1]
```diff
@@ -65,6 +69,43 @@ public void testGenericTableSchema() throws Exception {
...
+		catalog.createDatabase(db1, createDb(), false);
+		ObjectPath tablePath = new ObjectPath(db1, "generic110");
+
+		// create a table with old schema properties
+		Table hiveTable = org.apache.hadoop.hive.ql.metadata.Table.getEmptyTable(tablePath.getDatabaseName(),
+				tablePath.getObjectName());
+		// create table generic110 (c char(265), vc varchar(65536), ts timestamp(3), watermark for ts as ts)
+		hiveTable.setDbName(tablePath.getDatabaseName());
+		hiveTable.setTableName(tablePath.getObjectName());
+		hiveTable.getParameters().put(CatalogConfig.IS_GENERIC, "true");
```
Comment (JingsongLi) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogGenericMetadataTest.java:83]: Can we add tests for all types and watermarks and computed columns?

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogGenericMetadataTest.java [hunk 2]
```diff
@@ -65,6 +69,43 @@ public void testGenericTableSchema() throws Exception {
 		}
 	}
 
+	@Test
```
Comment (JingsongLi) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/catalog/hive/HiveCatalogGenericMetadataTest.java:72]: Can add a comments: `NOTE: Do not modify this test, it is important to forward compatibility.`.

**Comment by lirui-apache** [comment]
@JingsongLi I have added tests for more types and computed columns

**Pull request merged**